### ISSUE TITLE
when run as pre-commit hook ansible-lint silently fails to lint

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#611]
I discovered that ansible-lint when run as a pre-commit hook ansible-lint gives false positives, mainly because it fails to identify the files types.

I will give a censored example:

```
# outside pre-commit
$ ansible-lint  -v .                                                                                                                                                 
Examining /Users/ssbarnea/rdo/ansible-role-collect-logs/tasks/publish.yml of type tasks
/Users/ssbarnea/rdo/ansible-role-collect-logs/tasks/ara_graphite.yml:2: [E301] Commands should not change things if nothing needs doing

# with pre-commit (same version)
$ pre-commit run -v -a ansible-lint                                                                                                                                       [ansible-lint] Ansible-lint..............................................Passed
hookid: ansible-lint
Examining /Users/ssbarnea/rdo/ansible-role-collect-logs/tasks/collect.yml of type playbook
```

Because pre-commit call ansible-lint for each file, this not only makes the process much slower but worse it prevents ansible lint from discovering the file types correctly. You can see that his guess was a playbook when in fact it was tasks inside roles.

One think for sure calling `ansible-lint /Users/ssbarnea/rdo/ansible-role-collect-logs/tasks/ara_graphite.yml` confuses it to think it is a playbook, but allowing it to find files seems to be working.

One quick fix for this would be to disable the file parsing support in

**Comment by awcrosby** [comment]
@ssbarnea can you confirm this occurs with `v4.1.1a2` but not `v4.1.0`?

**Comment by ssbarnea** [comment]
@awcrosby I am almost sure this bug was there from start, we just failed to see that when run as a hook it was linting *(differently)* than directly.

**Comment by brucellino** [comment]
I landed here after I my CI was failing due to Ansible incorrectly linting my `.github/workflows` directory. Just wanted to say for the record that I implemented a workaround by passing `args` to `ansible-lint` instead of to `pre-commit`:

```yaml
---
repo:
  - repo: https://github.com/ansible-community/ansible-lint
    rev: v6.17.0
    hooks:
      - id: ansible-lint
        args: ["--exclude", ".github"]
```

**Comment by ssbarnea** [comment]
@brucellino I advice you to avoid adding any custom arguments inside pre-commit config because it will make the linter behave different when run outside pre-commit hook, from command line or from other tools like vscode extension. Instead add needed settings inside linter config.

**Comment by brucellino** [comment]
Excellent point, thank you!
